### DISCUSSION TITLE
Proper setting of device for pretrain model

### DISCUSSION THREAD

**Discussion by peiyaoli** [datamol-io/molfeat#49] [Ideas]
Hi, I tried to use the ChemGPT-1.2B model:

```Python
import datamol as dm
import os
from molfeat.trans.pretrained.hf_transformers import PretrainedHFTransformer

os.environ["TOKENIZERS_PARALLELISM"] = "false" # annoying huggingface warning
data = dm.freesolv().iloc[:1000]
transformer = PretrainedHFTransformer(kind="ChemGPT-1.2B", notation="selfies", device="cuda:0")
features = transformer(data["smiles"])
print(features.shape)
```
Now the output embedding dim is 2048 which means it is 1.2B model. But is the device set correctly?
I use command `nvidia-smi`  to check the GPU usage. But it looks like model hasn't loaded into GPU.

**Comment by cwognum** [comment]
Good catch! It seems the device is currently not actually used for `PretrainedHFTransformer`. Would you be willing to propose a fix? I'm happy to answer any questions you might have in the process.

**Comment by kkovary** [comment]
I think this should have been fixed by #88.
